### PR TITLE
feat: add service target fields support to aws module

### PR DESCRIPTION
**Pull request by kruskall**
Sets service target fields for services in the aws module,
namely s3 and dynamodb.

See https://github.com/elastic/apm/blob/main/specs/agents/tracing-instrumentation-db.md#aws-s3
See https://github.com/elastic/apm/blob/main/specs/agents/tracing-instrumentation-db.md#aws-dynamodb

This is part of the work to improve services dependencies granularity.

### CODE REVIEW

File: module/apmawssdkgo/sns.go [hunk 1]
```diff
@@ -59,6 +59,8 @@ func (s *apmSNS) spanName() string { return s.name }
 
 func (s *apmSNS) resource() string { return s.resourceName }
 
+func (s *apmSNS) targetName() string { return "" }
```
Comment (axw) [inline module/apmawssdkgo/sns.go:62]: https://github.com/elastic/apm/blob/main/specs/agents/tracing-instrumentation-messaging.md#span-context-fields says this should be the queue (topic) name.

File: module/apmawssdkgo/sqs.go [hunk 2]
```diff
@@ -72,6 +72,8 @@ func (s *apmSQS) spanName() string { return s.name }
 
 func (s *apmSQS) resource() string { return s.resourceName }
 
+func (s *apmSQS) targetName() string { return "" }
```
Comment (axw) [inline module/apmawssdkgo/sqs.go:75]: As above, this should be set to the queue name.

**Review by axw — COMMENTED** [review]
Looks good, may as well add the SNS/SQS target names while we're at it I think? Can you please add tests?

File: module/apmawssdkgo/sns.go (hunk 1 above)
Comment (kruskall) [reply module/apmawssdkgo/sns.go]: I missed that document! Thank you for pointing it out! :bow:

**Review by axw — APPROVED** [review]
LGTM!

**Pull request merged**